src/functions/certGenInit.ts

```diff
@@ -12,47 +18,55 @@ import { Utils } from "../utils/Utils";
  * @param callback - callback function
  */
 const certGenInit: Handler = async (
-  event: any,
+  event: DynamoDBStreamEvent,
   context?: Context,
   callback?: Callback
-): Promise<void | Array<SendMessageCommandOutput | any>> => {
+): Promise<DynamoDBBatchResponse> => {
   if (!event) {
     console.error("ERROR: event is not defined.");
-    return;
+    throw new Error("ERROR: event is not defined");
   }
 
-  // Convert the received event into a readable array of filtered test results
-  const expandedRecords: any[] = StreamService.getTestResultStream(event);
-  console.log(`Number of Retrieved records: ${expandedRecords.length}`);
-  const certGenFilteredRecords: any[] =
-    Utils.filterCertificateGenerationRecords(expandedRecords);
-
-  console.log(`Number of Filtered Retrieved Records: ${certGenFilteredRecords.length}`);
-
-  // Instantiate the Simple Queue Service
-  const sqService: SQService = new SQService(new SQSClient());
-  const sendMessagePromises: Array<
-    Promise<SendMessageCommandOutput | ServiceException>
-  > = [];
-
-  certGenFilteredRecords.forEach((record: any) => {
-    const stringifiedRecord = JSON.stringify(record);
-    console.log(stringifiedRecord);
-    sendMessagePromises.push(
-      sqService.sendCertGenMessage(stringifiedRecord)
-    );
-  });
-
-  return Promise.all(sendMessagePromises).catch((error) => {
-    console.error(error);
-    console.log("expandedRecords");
-    console.log(JSON.stringify(expandedRecords));
-    console.log("certGenFilteredRecords");
-    console.log(JSON.stringify(certGenFilteredRecords));
-    if (error.code !== "InvalidParameterValue") {
-      throw error;
+  const batchItemFailures: DynamoDBBatchItemFailure[] = [];
+  let expandedRecords: any[] = [];
+  let certGenFilteredRecords: any[] = [];
+
+  for (const record of event.Records) {
+    try {
+      expandedRecords = StreamService.getTestResultStream(record);
+      console.log(`Number of Retrieved records: ${expandedRecords.length}`);
+
+      certGenFilteredRecords =
+        Utils.filterCertificateGenerationRecords(expandedRecords);
+      console.log(
+        `Number of Filtered Retrieved Records: ${certGenFilteredRecords.length}`
+      );
+
+      // Instantiate the Simple Queue Service
+      const sqService: SQService = new SQService(new SQSClient());
```

Comment:
Are we better off hoisting this up above the `for ... of` loop to ensure it's only created once?

Reply:
yeah, thanks, silly error!